ARGO/src/Game.cpp

```diff
@@ -16,19 +17,25 @@ Game::Game() :
 		if (SDL_Init(SDL_INIT_EVERYTHING) < 0) throw "Error Loading SDL";
 
 		// Create SDL Window Centred in Middle Of Screen
-		m_window = SDL_CreateWindow("Final Year Project", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, 1920, 1080, NULL);
+		m_window = SDL_CreateWindow("ARGO", SDL_WINDOWPOS_CENTERED, SDL_WINDOWPOS_CENTERED, 1920, 1080, NULL);
+		
 		// Check if window was created correctly
 		if (!m_window) throw "Error Loading Window";
 
 		//Create the SDL Renderer 
 		m_renderer = SDL_CreateRenderer(m_window, -1, 0);
-		//Check if the renderer was created correclty
+		//Check if the renderer was created correctly
 		if (!m_renderer) throw "Error Loading Renderer";
 
 		// Sets clear colour of renderer to black and the color of any primitives
 		SDL_SetRenderDrawColor(m_renderer, 0, 0, 0, 255);
 		// Game is running
 		m_isRunning = true;
+		for (int index = 0; index < SDL_NumJoysticks(); index++)
+		{
+			controllers.push_back(new Controller());
```

Comment:
You can reduce the amount of calls by doing controllers.emplace_back().
This will allow for not creating the controllers on the heap, preventing memory leaks.
This calls the constructor and puts it directly into the vector rather than creating it, copying it and then destroying it.